### PR TITLE
Add Azure Devops Project

### PR DESCRIPTION
**Pull request by shep1987**
Looking for feedback on the first steps of adding Azure Devops Support

Added `ISettingsReader` needs to be implemented by each platform, this gets the Repository and Auth settings for it
Commands have this injected into the constructor
Added tests around this for the GitHub Implementation (`GitHubSettingsReader`)

`IForkFinder` become something that each platform must implement 
Azure supports only `ForkMode.SingleRepositoryOnly` at the moment

`IRepositoryDiscovery` become something that each platform must implement 
Azure supports only `ServerScope.Repository` at the moment

Added most of the implementation to `AzureDevOpsPlatform` using a rest client but this will need further work to make it more efficient and improve the structure but it does work.

Removed/Renamed some GitHub Specific language.

### CODE REVIEW

File: NuKeeper.Abstractions/Configuration/AuthSettings.cs [hunk 1]
```diff
@@ -12,5 +12,6 @@ public AuthSettings(Uri apiBase, string token)
 
         public Uri ApiBase { get; }
         public string Token { get; }
+        public string AzureDevOpsOrganisation { get; set; }
```
Comment (skolima) [inline NuKeeper.Abstractions/Configuration/AuthSettings.cs]: abstraction leaking?

Comment (shep1987) [reply NuKeeper.Abstractions/Configuration/AuthSettings.cs]: Yup, nice catch fix pushed

**Comment by AnthonySteele** [comment]
Seems OK to me. Is this good to merge now?

**Comment by shep1987** [comment]
I am happy with it

**Pull request merged**